Gateway/Request/MerchantAccountDataBuilder.php

```diff
@@ -52,8 +53,9 @@ public function build(array $buildSubject)
         $paymentDataObject = \Magento\Payment\Gateway\Helper\SubjectReader::readPayment($buildSubject);
         $order = $paymentDataObject->getOrder();
         $storeId = $order->getStoreId();
+        $payment = $paymentDataObject->getPayment();
 
-        $merchantAccount = $this->adyenHelper->getAdyenAbstractConfigData("merchant_account", $storeId);
+        $merchantAccount = $this->adyenHelper->getAdyenMerchantAccount($payment, $storeId);
```

Comment:
You know send the whole payment object to the method where you only need:
$payment->getMethod() so just sent this into the call.